### ISSUE TITLE
Chore: Clean up code, Java conventions, and mapping conventions.

### ISSUE THREAD

**Issue by bconlon1** [Gilded-Games/The-Aether#311]
- Keep methods in fairly consistent order.
- Update parameters and variable names, make sure they're camelcase.
- Cleanup unused imports (use IntelliJ's code analyze option).
- Fix any spacing, unnecessary returns (also as in spacing), bracket placement, lack of usage of `this.`
- Use getters and setters and keep fields private for new fields.
- Try to use getters and setters for vanilla fields regardless of protection.
- Any usage of `@SupressWarnings` should probably have documentation detailing why they're there if more are put in. 
- Properly annotate with `@Override`.
- Look for ways to abstract code into functional interfaces in instances of repeated code.
- Abstract some worldgen (trees) and rendering stuff (multi model renderers) to Nitrogen

**Comment by bconlon1** [comment]
Refactoring to `core.mixin` pending/in-progress with #310.

**Comment by bconlon1** [comment]
`core.registry` and `core.api` are good.

**Comment by bconlon1** [comment]
Refactoring to `core.capability`, `core.network`, `core.util` (except BlockPlacers), and `core.data` (except AetherWorldProvider) pending with #316.

**Comment by bconlon1** [comment]
`core` package should be reviewed again for certain mapping-related refactors that have to be done still.

**Comment by bconlon1** [comment]
~~Currently blocked by #518.~~

**Comment by bconlon1** [comment]
Progress:
- [x] `advancement`
- [x] `api`
- [x] `block`
- [x] `blockentity`
- [x] `capability`
- [x] `client`
- [x] `command`
- [x] `data`
- [x] `effect`
- [x] `entity`
- [x] `event`
- [x] `integration` 
- [x] `inventory`
- [x] `item`
- [x] `loot`
- [x] `mixin`
- [x] `network`
- [x] `perk`
- [x] `recipe`
- [x] ~~`util`~~
- [x] `world` 
- [x] Aether
- [x] AetherConfig
- [x] AetherTags

**Comment by bconlon1** [comment]
~~Block and Item refactoring pending on https://github.com/MinecraftForge/MinecraftForge/pull/8838.~~

**Comment by bconlon1** [comment]
Various item/block fixes and refactors that need doing or looking into:
- [x]  Use more `<br>`s for javadoc readability. 
- [x] More capitalization for items in javadocs 
- [x] Make sure param javadocs link class.
- [x] Abstract full swing check to EquipmentUtil.
![image](https://user-images.githubusercontent.com/67203206/190957801-82c51e38-bb76-465e-8854-28366777d545.png)
- [x]  Item repair ingredient tags.
- [x]  Abilities need to be cleaned up to no longer be tag and mixin-based because the implementation was too rough. 
- [x]  Remove the weapon and tool tags that were previously used for basing behavior off of because they're redundant at this point.
- [x]  Gravitite Sword ability should only work if the mob is on the ground.
- [x]  Abilities should only occur when an attack is at full strength. 
- [x]  Rename lol_us kino stick to bdogz stick
- [x]  Zombified Piglins for some reason don't target the player when killed with a Pig Slayer.
- [x]  Pig Slayer should double cooked porkchops in addition to raw porkchops.
- [x]  Potentially balance Pig Slayer to not one-shot Piglin Brutes or Hoglins but maybe just two-shot them.
- [x]  Reorganize `item.combat` package a bit.
- [x]  Unique lore entry for jeb hammer if doable. Might just need a custom hardcoded override. Maybe have a list for this for modders to be able to use to inject override entries to the book of lore, since this wouldn't be able to be utilized in a non-addon context.
- [x]  Valkyrie Tools' extended reach needs to be made to not work if in the offhand or any other situation where it can be exploited. https://github.com/Gilded-Games/The-Aether/pull/521/commits/65c4d321d0c00c4e5e67a0f41717a807601bc71a
- [x]  Valkyrie Tools and Lance may need slower attack speed.
- [x]  Darts will float in the air if you shoot a lot at an entity at a time. Might be caused by invincibility frames. If deflected off an entity they should regain gravity.
- [x]  .orElse(null) check needs to be removed within the dart layer renderers or anywhere else and be replaced with an ifPresent() lambda.
- [x]  Obsidian Armor needs its ability removed; it didn't have one afaict.
- [x]  Dungeon Armor gloves' attack strength need rebalancing to match Netherite Gloves.
- [x]  Glove attack strength might need more balancing again.
- [x]  Ice accessories need to be able to "freeze" waterlogged blocks like Icestone does at the moment.
- [x]  Config option for ice accessories to create temporary blocks instead of permanent blocks. Look at conditional Forge recipes.
- [x] Document FreezingBehavior
- [x]  Need to add cauldron interactions for the Skyroot Powder Snow Bucket
- [x]  Aether Portal Item placement needs to not be blocked by grass and snow, it should replace those upon placement.
- [x]  Keys need a better handheld model.
- [x]  Remove any textures from the optional resource packs if they're the same as the new textures to cut down on space.
- [x] Fix Hammer of Notch projectile being able to push entities like the Sun Spirit. 
- [x] Cape rotation is broken on multiplayer; need a packet that sends the data to all clients not just the player with the cape.
- [x] Valkyrie Armor wing rotation continues even when the game is paused.
- [x] Our unique Curio slot tags should probably by default include some of the standardized Curio tags so other mods' Curios can be used by default in our slots.
- [x] Configs for Hammer of Notch and Cloud Staff cooldowns.
- [x] Rotate Victory Medal held model.
- [x] In Survival, gloves still render in first person even when using the invisibility cloak. They also don't render with the invisibility effect even though they should because it doesn't remove armor in third person.
---
- [x] Gravitite Axe causes floated blocks to spawn two blocks from it when colliding.
- [x] Gravitite Pickaxe doesn't work.
- [x]  Double drop related classes need cleanup and I need to get rid of the DoubleDrops interface.
- [x]  FrictionCapped needs to be fixed to work for non-player entities by being made to run on both level sides.
- [x]  It might make sense to make Ambrosium Blocks a fuel source, as well as icestone walls, stairs, and slabs.
- [x]  See if it's possible to make tall grass on top of enchanted aether grass match the color, if it is not too much of a hassle to get working (see ColorResolver)
- [x]  Drop Golden Apples with high rarity from Golden Oak Leaves. Reimplement relevant pro tip.
- [x] Improve dungeon block overlay performance.
- [x] Improve dungeon block overlay appear and disappear times.
- [x] Let Enchanted Grass be bonemealable. But don't let it spread.
- [x] Mossy Holystone recipe with vines/moss + holystone.
- [x] Cleanup AetherTeleporter.
- [x] The disable_aether_portal config doesn't work or do anything and needs fixing.
- [x] Some way of letting entities not trigger trapped blocks.
- [x] Remove onlyin annotations.
- [x] Stripped blocks need to not use the double drops function for their loot tables (currently it makes them drop nothing).
---
- [x] Rename DungeonTracker to BossRoomTracker
- [x] Maybe it'd be worth ensuring some data is tracked on the client with BossRoomTracker but is only queried on the server by the mod. In case addons for some reason need the info on the client even though we don't.
- [x] Sun Spirit can be pushed with explosions.
- [x] ValkyrieWingsLayer needs a parameter for the texture; Valkyrie Queens are meant to have a different wing texture but they use the regular valkyrie wing texture instead.
- [x] Valkyrie Queen can sometimes still teleport outside of the dungeon room bounds.
- [x] Weird slider shield breaking behavior. Could maybe get around this with cancelling ShieldBlockEvent for the Slider.
- [x] Possibly make snowballs damage Fire Minions.
- [x] There may be a bug where you're stopped from being eaten by Swets if you crouch.
- [x] Frogs try to eat swets being ridden by players; maybe cancel frogs targeting swets in this scenario.
- [x] ContinuousAttackGoal for Mimics?
- [x] Aerbunnies aren't immediately able to immediately use their jump boost when falling; I think this was intentional but the delay between reaching max height when jumping and falling a speed enough to let them jump is slightly too much at the moment.
- [x] Swets need to not use the Slime Squishes subtitle.
- [x] Swets need to not spawn slime particles.
- [x] Aerbunnies need to not be able to be interacted with by other players in mobInteract if they're mounted.
- [x] Aerbunnies need to automatically demount on both sides if a player goes into spectator. 
---
- [x] Certain gamerules like randomTickSpeed can't be customized in the Aether (changing them doesn't reflect in the Aether but does in the overworld).
- [x] Sleeping in a bed changes the time with overworld time intervals and not aether time intervals.
- [x] Check if certain GUI elements like the "Ascending to the Aether" text properly respect GUI sizing settings and if they may need repositioning in these situations.
- [x] Halo perk doesn't follow crouch offset.
- [x] Aether Portal can replace locked blocks if it spawns in a dungeon.
---
- [x] Fix void fog.
- [x] Some block hardnesses are not accurate to the original mod.
- [x] Cockatrice darts don't damage shields, other projectiles might have a similar issue.
- [x] Did moas have rarity based on color for spawning in beta.
- [x] Trapped blocks should spawn mobs based on the player's facing direction. - Cactus
- [x] Reduce trapped block volume. - Cactus
- [x] Prevent portals from spawning on blue aerclouds
- [x] Potentially weird behavior with Swets duplicating on death even when they shouldn't.
- [x] Swets aren't dropping experience, and most mobs it turns out.
- [x] Reduce spawnrate of swets, whirlwinds, aechor plants; reduce swet pack size.
- [x] Increase gravitite rates.
- [x] Larger swet hitbox
- [x] Sentry hitbox is inaccurate; render is a size bigger than its assigned slime size, which is intended, however it means the hitbox is one size smaller than the render size so the hitbox needs to be tweaked.
- [x] Sentries can explode even after death if collided with.
- [x] Baby Moas shouldn't be able to spawn naturally.
- [x] Swets too common, aechor plants too common, whirlwinds not common enough, zephyrs not common enough.
- [x] Prevent cockatrices from spawning in rain.
- [x] Fix custom advancement jingles for bosses.
- [x] Discuss balance of having hostiles target passives.
- [x] Increase minimum spacing of aerial dungeons.
- [x] Blue name for frosted disc
- [x] swet can still duplicate
- [x] Whirlwind unaffected mob tag
- [x] Incorrect phyg texture
- [x] gravitite can be broken with torches; only break floating blocks maybe if theyre by-hand harvest level
- [x] enchanted berries need more saturation
- [x] leftover bucket issue with altars?
- [x] eternal day syncing issue on servers - Cactus
- [x] tall grass on by default
- [x] edible ambrosiun on by default
- [x] Recipe for shield in aether
- [x] balance moa jump power
- [x] nerf aerbunnies
- [x] nerf swet health
- [x] remove pink aerclouds - Bconlon
- [x] fix gravitite dupe bug
- [x] mimics need shorter attack range - Cactus
- [x] Subtitles don't use aether namespace. - Bconlon
- [x] Subtitles call Aether crafting stations furnaces; not sure if they should even use furnace noises. - Bconlon
- [x] disc - Bconlon
- [x] Moa type render crash; add default values to texture locations in case of nullability. - Drullkus 
- [x] Slider can't target the player if they're wearing an invisibility cloak. Should add a boss tag check for the visibility blocking.
- [x] another grav dupe bug
- [x] Simplify check here https://github.com/Gilded-Games/The-Aether/blob/ab3df95e491d77434f87f022a4e4468ed1be5476/src/main/java/com/gildedgames/aether/block/miscellaneous/FloatingBlock.java#L54 - Bconlon
- [x] make ambrosium furnace fuel - bconlon
- [x] Spread out bronze dungeons a bit more again due to overlaps - Oz
- [x] review mob chase ranges like for cockatrices and zepyhr - Oz 
- [x] gravitite ore drops when reaching height limit
- [x] moa flap sound isn't playing
- [x] moas need step sound + faster flap animation(?) - Bconlon
- [x] moa flap slightly jitters when landing?
- [x] moa can flap when sitting: https://discord.com/channels/118816101936267265/932492946052362240/1077680231948767234
- [x] other weird things can happen with nature staffed moas: https://discord.com/channels/118816101936267265/932492946052362240/1077680764570828871
- [x] bug with aerbunny and blue aercloud interaction
- [x] Tall grass doesn't spawn under terrain; may need to use a different heightmap or the improved layer placement modifier. Could trees also do the heightmap thing? - bconlon
- [x] Better healing/inebriation overlay textures - Oz 
- [x] Orange moa balancing or removal  
- [x] Update curios dependency
- [x] Double check that all lore entries are actually accurate to default item states and their current iterations in the mod. - Bconlon
- [x] more advancements and new layout - Oz
- [x] opaquer aerclouds - Oz
- [x] Crash when other entities mount moas? - bconlon
- [x] rebalance obsidian armor obtainmentm - Bconlon
- [x] include trigger for inventory_changed in the obsidian armor conversion code - Bconlon
- [x] cold lightmap - bconlon
- [x] green sunset config  - bconlon
- [x] lance mending fix? - bconlon  
- [x] more wool usable in blue cape
- [x] sentry slime issues - Oz
- [x] sentry reach - Programmer
- [x] zanite accessory durability desig - Programmer
- [x] Valkyrie Queen can despawn when chunks are unloaded - Programmer
- [x] Crystal projectiles need the spawn particle code coupled with their discard code to ensure the particles always spawn with projectile death - Programmer
- [x] ice ball balancing again - Oz 
- [x] Sun Spirit doesn't face player when invisible - Programmer
- [x] Moas can age up without being fed; might be something from the AgeableMob code interfering - Programmer
- [x] adjust sunset color - Oz/Programmer
- [x] missing cloud crystal death lang and also cloud parachute pro tip should say cold parachute - Programmer
- [x] treasure chest lore fix - Oz/Programmer
- [x] look into balancing crafter recipe speeds - Oz  
- [x] correct zanite/gravitite textures in old packs - Oz
- [x] aechor plant spawn changes - Oz/Programmer

**Comment by bconlon1** [comment]
~~New PR to keep up with: https://github.com/MinecraftForge/MinecraftForge/pull/9000~~

**Comment by bconlon1** [comment]
~~Relevant to block cleanup: https://github.com/MinecraftForge/MinecraftForge/pull/9139~~
